src/System.Drawing.Common/src/misc/DbgUtil.cs

```diff
@@ -65,117 +51,14 @@ public static void AssertFinalization(object obj, bool disposing)
         }
 
         [Conditional("DEBUG")]
-        public static void AssertWin32(bool expression, string format, object arg1)
-        {
-#if DEBUG
-            if (!expression)
-            {
-                object[] args = new object[] { arg1 };
-                AssertWin32Impl(expression, format, args);
-            }
-#endif
-        }
-
-        [Conditional("DEBUG")]
-        public static void AssertWin32(bool expression, string format, object arg1, object arg2)
-        {
-            if (!expression)
-            {
-                object[] args = new object[] { arg1, arg2 };
-                AssertWin32Impl(expression, format, args);
-            }
-        }
-
-        [Conditional("DEBUG")]
-        public static void AssertWin32(bool expression, string format, object arg1, object arg2, object arg3)
-        {
-            if (!expression)
-            {
-                object[] args = new object[] { arg1, arg2, arg3 };
-                AssertWin32Impl(expression, format, args);
-            }
-        }
-
-        [Conditional("DEBUG")]
-        public static void AssertWin32(bool expression, string format, object arg1, object arg2, object arg3, object arg4)
-        {
-            if (!expression)
-            {
-                object[] args = new object[] { arg1, arg2, arg3, arg4 };
-                AssertWin32Impl(expression, format, args);
-            }
-        }
-
-        [Conditional("DEBUG")]
-        public static void AssertWin32(bool expression, string format, object arg1, object arg2, object arg3, object arg4, object arg5)
-        {
-            if (!expression)
-            {
-                object[] args = new object[] { arg1, arg2, arg3, arg4, arg5 };
-                AssertWin32Impl(expression, format, args);
-            }
-        }
-
-        [Conditional("DEBUG")]
-        private static void AssertWin32Impl(bool expression, string format, object[] args)
+        public static void AssertWin32(bool expression, string format, params object[] args)
         {
             if (!expression)
             {
+                var e = new Win32Exception();
```

Comment:
Should this variable be behind \#if DEBUG, since it's only used in a Debug.Fail message?

Reply:
The whole method is `[Conditional("DEBUG")]`

Reply:
I was going to say "but if the caller was DEBUG and this was not it would be different".  Then I noticed the type was internal, so that'd be why it's not an issue.

So the type will still exist in the release build, and the IL will be there to call Win32Exception(), but there will be no callers (assuming no InternalsVisibleTo).  Okay, not worth changing :smile:.

Reply:
> So the type will still exist in the release build

I'd have to check, but I would hope in a release build the linker would remove the type in its entirety, as no one will be referencing it.